src/core/geometry/qgsorientedbox3d.cpp

```diff
@@ -149,3 +173,72 @@ QgsOrientedBox3D QgsOrientedBox3D::transformed( const QgsMatrix4x4 &transform )
                            << col2.x() << col2.y() << col2.z()
                            << col3.x() << col3.y() << col3.z() );
 }
+
+bool QgsOrientedBox3D::intersects( const QgsOrientedBox3D &other ) const
+{
+  // use the Separating Axis Theorem (SAT) for OBB (Oriented Bounding Box) collision detection.
```

Comment:
It may be worth adding reference to the original paper which gives the reasoning and formulas:
section 5 in `OBBTree: A Hierarchical Structure for Rapid Interference Detection` (1996)

Reply:
done